src/templates/ByteArrayHelper.java

```diff
@@ -0,0 +1,26 @@
+package {{ config.package_name() }};
+
+import java.nio.ByteBuffer;
+
+public enum FfiConverterByteArray implements FfiConverterRustBuffer<byte[]>{
+  INSTANCE;
+
+    @Override
+    public byte[] read(ByteBuffer buf) {
+        int len = buf.getInt();
+        byte[] byteArr = new byte[len];
+        buf.get(byteArr);
+        return byteArr;
+    }
+  
+    @Override
+    public long allocationSize(byte[] value) {
+        return 4L + (long)value.length;
```

Comment:
length always 32 bits, even on 64 bit architectures? Looks like Java doesn't have `sizeof()`, so doing something different would probably suck anyway.

Reply:
The return value from Java for `length` is always an `int` regardless of arch/JVM. The actual max size of the array is JVM/arch dependant, but pretty sure it's always within int max.